server/leafnode.go

```diff
@@ -1292,10 +1285,49 @@ func (c *client) processLeafSub(argo []byte) (err error) {
 	key := string(sub.sid)
 	osub := c.subs[key]
 	updateGWs := false
+	acc := c.acc
 	if osub == nil {
+		subj := string(sub.subject)
+		accUnlock := false
```

Comment:
For code clarity, I would have probably locked unconditionally, but I understand why you do it this way, and it will reduce contention.